point.go

```diff
@@ -24,6 +26,22 @@ func NewPointFlat(l Layout, flatCoords []float64) *Point {
 	return g
 }
 
+// NewPointFlatMaybeEmpty returns a new point, checking whether the point may be empty
+// by checking wther all the points are NaN.
+func NewPointFlatMaybeEmpty(layout Layout, flatCoords []float64) *Point {
+	isEmpty := true
+	for _, coord := range flatCoords {
+		if !math.IsNaN(coord) {
```

Comment:
Here the code use `math.IsNaN`, but really to represent an empty point we're using a specific NaN value for all elements of `flatCoords`, so this test is a little too eager to set `isEmpty` to `true`. I would replace it with a test that checks that all elements of `flatCoords` are the specific NaN that we use.

Reply:
we can't test that as NaN != NaN :(

Reply:
(well, without doing some complex hex stuff: https://stackoverflow.com/questions/21869571/nan-different-from-nan-in-go)

Reply:
It is possible: https://play.golang.org/p/w7Pov-SPuSJ